ext/strings.go

```diff
@@ -250,19 +250,62 @@ const (
 //
 //	'TacoCat'.upperAscii()      // returns 'TACOCAT'
 //	'TacoCÆt Xii'.upperAscii()  // returns 'TACOCÆT XII'
-func Strings() cel.EnvOption {
-	return cel.Lib(stringLib{})
+func Strings(options ...StringsOption) cel.EnvOption {
+	s := &stringLib{}
+	for _, o := range options {
+		s = o(s)
+	}
+	return cel.Lib(s)
 }
 
-type stringLib struct{}
+type stringLib struct {
+	locale string
+}
 
 // LibraryName implements the SingletonLibrary interface method.
 func (stringLib) LibraryName() string {
 	return "cel.lib.ext.strings"
 }
 
+// StringsOption is a functional interface for configuring the strings library.
+type StringsOption func(*stringLib) *stringLib
+
+// StringsLocale configures the library with the given locale. The locale tag will
+// be checked for validity at the time that EnvOptions are configured. If this option
+// is not passed, string.format will behave as if en_US was passed as the locale.
+func StringsLocale(locale string) StringsOption {
+	return func(sl *stringLib) *stringLib {
+		sl.locale = locale
+		return sl
+	}
+}
+
 // CompileOptions implements the Library interface method.
-func (stringLib) CompileOptions() []cel.EnvOption {
+func (sl stringLib) CompileOptions() []cel.EnvOption {
+	formatOverload := cel.MemberOverload("string_format", []*cel.Type{cel.StringType, cel.ListType(cel.DynType)}, cel.StringType,
+		cel.FunctionBinding(func(args ...ref.Val) ref.Val {
+			s := args[0].(types.String).Value().(string)
+			formatArgs := args[1].(traits.Lister)
+			return stringOrError(stringFormat(s, formatArgs))
+		}))
+	if sl.locale != "" {
```

Comment:
You could consider using a local variable assigned to the defaultLocale, and then testing if the optional locale is set, parsing it, and returning an error or updating the local variable ... this would allow you to provide the local variable to a single `stringFormatWithLocale` overload and preserve the existing behavior.

Reply:
Gotcha, fixed